frmts/kea/keaband.cpp

```diff
@@ -558,6 +558,44 @@ double KEARasterBand::GetNoDataValue(int *pbSuccess)
     }
 }
 
+int64_t KEARasterBand::GetNoDataValueAsInt64(int *pbSuccess)
+{
+    try
+    {
+        int64_t nVal;
+        this->m_pImageIO->getNoDataValue(this->nBand, &nVal, kealib::kea_64int);
+        if( pbSuccess != nullptr )
+            *pbSuccess = 1;
+
+        return nVal;
+    }
+    catch (const kealib::KEAIOException &)
+    {
+        if( pbSuccess != nullptr )
+            *pbSuccess = 0;
+        return -1;
+    }
+}
+
+uint64_t KEARasterBand::GetNoDataValueAsUInt64(int *pbSuccess)
+{
+    try
+    {
+        uint64_t nVal;
+        this->m_pImageIO->getNoDataValue(this->nBand, &nVal, kealib::kea_64uint);
+        if( pbSuccess != nullptr )
+            *pbSuccess = 1;
+
+        return nVal;
+    }
+    catch (const kealib::KEAIOException &)
+    {
+        if( pbSuccess != nullptr )
+            *pbSuccess = 0;
+        return -1;
```

Comment:
to avoid confusion with a negative value in a signed type.
```suggestion
        return std::numeric_limits<uint64_t>::max();
```
you'll also need to add a ``#include <limits>``

Reply:
Good call, I've just committed this change.